javaparser-core/src/main/java/com/github/javaparser/ast/body/EnumDeclaration.java

```diff
@@ -140,7 +141,7 @@ public EnumDeclaration setImplementedTypes(NodeList<ClassOrInterfaceType> implem
     }
 
     public EnumConstantDeclaration addEnumConstant(String name) {
-        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(assertNotNull(name));
+        EnumConstantDeclaration enumConstant = new EnumConstantDeclaration(name);
```

Comment:
Hmmm, see it as a method precondition: parameter name should not be null. It is definitely not required, but it makes the stack trace clearer: it points to this location instead of the constructor of SimpleName. The user will then have to figure out why that constructor is called at all, and deduce from that nullpointer that they shouldn't have passed in a null String.

I say leave it in, the more asserts the merrier :)